Added API and new command line parameters

anthonymorast opened the pull request.

testhound reviewed (changes requested):
1. Can you delete the KC_RAD_Hourly.txt.timeframe_* files? These files are auto generated.
2. Can you delete the PalValidator executable? I don't want the executable as part of the repo.
3. Can you leave support for other historical file formats in getHistoricDataFileReader? I may want to use other csv file formats when running on local files
4. I thought we were going to have a separate file for the "api" option where the keys for the different API formats are stored. It looks like right now the key is in the second config file.

anthonymorast commented:
1, 2 4. Yeah sorry about that. I wasn't actually using the examples/KC directory. I moved the example files into my CMake "build" directory to run the examples right after building the project rather than copying the executable so the examples/KC/ directory is a little behind the actual implementation. I copied the files over to the examples directory and cleaned it up.

3. Re-implemented this functionality and added the file format back to config 1.

testhound reviewed (approved):
Approved/
